### PR TITLE
Promote ir namespace and create draft libspirv.{h|c}pp.

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: source/opt/libspirv.hpp [hunk 1]
```diff
@@ -24,31 +24,35 @@
...
-#include "source/opt/module.h"
+#include "module.h"
+#include "spirv-tools/libspirv.h"
 
 namespace spvtools {
-namespace opt {
 
-// TODO(antiagainst): expand and export these functions as the C++ interface in
-// libspirv.hpp.
+// TODO(antiagainst): add parameters for target environment and error message.
```
Comment (dneto0) [inline source/opt/libspirv.hpp]: Let's add the context object before making this API public.
I think it should be spvtools::ir::Context  and then the Assemble, Disassemble, and BuildModule functions are methods on the context object.

**Comment by dneto0** [comment]
Please add the context object now.

File: source/opt/libspirv.hpp (hunk 1 above)
Comment (antiagainst) [inline source/opt/libspirv.hpp]: Wrapped all functions in a class.

But I feel like `Context` is too vague as a name. I'd prefer `SpvTools`. It's much clear when reading together with, like `SpvTools.Assemble`.

**Comment by dneto0** [comment]
Please add a test that shows failure of the assembler on some source program that uses a 1.1 feature.  For example, OpSizeOf is new in 1.1

**Comment by antiagainst** [comment]
@dneto0: done.

**Comment by dneto0** [comment]
+2
Please squash at least the first two commits together.

**Pull request merged**